### PR TITLE
Fix Confusing Filter controls

### LINKED ISSUE #389: Filters: Confusing controls
@sammyd 

Do you find the filter controls 'clear all' and 'close' confusing?

Could we make them match the website instead (without icons). 

Screenshot 2020-02-25 at 10 39 22

### PR DESCRIPTION
**Pull request by byaruhaf**
Changed the Filter controls to match the RW website.

Rename
- "Clear All" => "Reset filters"
- "Close" => "Apply filters"

This PR fixes the issue's #416 & #389

### CODE REVIEW

**Comment by 0xTim** [comment]
Hey @byaruhaf thanks for the PR! I've seen it and will take a look tomorrow morning, but looks good at an initial glance!

File: Emitron/Emitron/Constants.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ import CoreGraphics.CGBase
 
 enum Constants {
   static let filters = "Filters"
-  static let clearAll = "Clear All"
+  static let resetFilters = "Reset filters"
```
Comment (0xTim) [inline Emitron/Emitron/Constants.swift]: Nit - `filters` should be capitalised
```suggestion
  static let resetFilters = "Reset Filters"
```

File: Emitron/Emitron/UI/Library/Filtering/AppliedFilterTagButton.swift [hunk 2]
```diff
@@ -120,7 +120,7 @@ struct AppliedFilterView_Previews: PreviewProvider {
   
   static var tags: some View {
     HStack {
-      AppliedFilterTagButton(name: "Clear All", type: .destructive) { }
+      AppliedFilterTagButton(name: "Reset filters", type: .destructive) { }
```
Comment (0xTim) [inline Emitron/Emitron/UI/Library/Filtering/AppliedFilterTagButton.swift]: Again nit - filters should be capitalised

```suggestion
      AppliedFilterTagButton(name: "Reset Filters", type: .destructive) { }
```

File: Emitron/Emitron/UI/Library/Filtering/FiltersView.swift [hunk 3]
```diff
@@ -70,7 +70,7 @@ struct FiltersView: View {
       
       HStack {
         
-        MainButtonView(title: "Clear All", type: .secondary(withArrow: false)) {
+        MainButtonView(title: "Reset filters", type: .secondary(withArrow: false)) {
```
Comment (0xTim) [inline Emitron/Emitron/UI/Library/Filtering/FiltersView.swift]: Nit - filters should be capitalised

```suggestion
        MainButtonView(title: "Reset Filters", type: .secondary(withArrow: false)) {
```

File: Emitron/emitronScreenshots/EmitronScreenshots.swift [hunk 4]
```diff
@@ -42,7 +42,7 @@ class EmitronScreenshots: XCTestCase {
     
     // Reset filters
     app.buttons.matching(identifier: "Filter Library").element.tap()
-    app.buttons.matching(identifier: "Clear All").element.tap()
+    app.buttons.matching(identifier: "Reset filters").element.tap()
```
Comment (0xTim) [inline Emitron/emitronScreenshots/EmitronScreenshots.swift]: Nit - filters should be capitalised

```suggestion
    app.buttons.matching(identifier: "Reset Filters").element.tap()
```

File: Emitron/Emitron/UI/Library/Filtering/FiltersView.swift [hunk 5]
```diff
@@ -106,8 +106,7 @@ struct FiltersView: View {
   }
   
   private func applyOrCloseButton() -> MainButtonView {
-    let equalSets = Set(libraryRepository.nonPaginationParameters) == Set(filters.appliedParameters)
-    let title = equalSets ? "Close" : "Apply"
+    let title = "Apply filters"
```
Comment (0xTim) [inline Emitron/Emitron/UI/Library/Filtering/FiltersView.swift]: Nit - filters should be capitalised

```suggestion
    let title = "Apply Filters"
```

File: Emitron/Emitron/UI/Library/Filtering/FiltersView.swift [hunk 6]
```diff
@@ -106,8 +106,7 @@ struct FiltersView: View {
   }
   
   private func applyOrCloseButton() -> MainButtonView {
```
Comment (0xTim) [inline Emitron/Emitron/UI/Library/Filtering/FiltersView.swift]: Sine this button no longer changes the title between apply or close, it should probably be renamed `applyFiltersButton()`

**Review by 0xTim — CHANGES_REQUESTED** [review]
Hey @byaruhaf - thanks for the great PR! Just a few small minor nits and we're good to go! 🎉

File: Emitron/Emitron/UI/Library/Filtering/FiltersView.swift (hunk 6 above)
Comment (byaruhaf) [reply Emitron/Emitron/UI/Library/Filtering/FiltersView.swift]: Hey @0xTim  thanks for the comments will make another PR for changing the name of function.

**Review by 0xTim — APPROVED** [review]
LGTM

**Comment by 0xTim** [comment]
@sammyd do we have any CI for the iOS app to run on PRs?

**Review by byaruhaf — COMMENTED** [review]
updated the requested changes

**Comment by sammyd** [comment]
> @sammyd do we have any CI for the iOS app to run on PRs?

@0xTim oh, we should yes....it has run SwiftLint, but it's supposed to run this too:

https://github.com/razeware/emitron-iOS/blob/development/.github/workflows/run_tests.yml

However, I've just noticed that it is only configured to run on push, not on PR. Should update that...

**Comment by 0xTim** [comment]
@sammyd I've added that in #466

**Comment by 0xTim** [comment]
@byaruhaf since you're the first public PR you're finding some of our teething issues that we need to fix! 😁 

The main one is that PRs aren't tested on CI. I've fixed that now on `development` so we need to merge those changes in to this PR. Could you rebase your fork onto upstream?

If you don't know how to do this, in Terminal, run:

```bash
# Add upstream if it's not already defined
git remote add upstream git@github.com:razeware/emitron-iOS.git
# Pull in all the changes from upstream
git fetch upstream
# Rebase your changes onto the existing changes
git rebase upstream/development
# Push your changes (needs a force since you've rewritten history with the rebase, but it keeps the history clean and is fine in this instance since it's your fork)
git push -f
```

**Comment by byaruhaf** [comment]
Let me get that done

**Comment by 0xTim** [comment]
Aha failing because it doesn't have access to download the secrets. @sammyd I'm thinking we merge this PR then fix CI in a separate PR to use dummy secrets for the tests and the real ones for building for Testflight/App Store. Thoughts?

**Comment by sammyd** [comment]
@0xTim this is the cause of the problem here:

image

As part of the workflow for `Run Tests`, we try to pull the secrets. They're not required for testing, so I think replacing this section of `.github/workflows/run_tests.yaml`:

```yaml
    - name: Download secrets
      env:
        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
      run: |
        scripts/download_s3.sh development > Emitron/Emitron/Configuration/secrets.development.xcconfig
        scripts/download_s3.sh beta > Emitron/Emitron/Configuration/secrets.beta.xcconfig
        scripts/download_s3.sh production > Emitron/Emitron/Configuration/secrets.production.xcconfig
```

with something like this:

```yaml
    - name: Generate secrets
      run: |
        scripts/generate_secrets.sh
```

might well do the trick.

I'm happy for this to be merged, and then fixed in a new PR. Sorry for not realised that secrets weren't going to be available as part of the normal OSS flow—it's pretty obvious now I think about it =)

**Pull request merged**